src/bit_operations.jl

```diff
@@ -31,16 +26,18 @@ bitarray(v::Number, nbits::Int)::BitArray{1} = vec(bitarray([v], nbits))
 bitarray(nbits::Int) = x -> bitarray(x, nbits)
 
 """
-    basis([IntType], nbits::Int) -> UnitRange{IntType}
-    basis([IntType], state::AbstractArray) -> UnitRange{IntType}
+    bit_basis([IntType], nbits::Int) -> UnitRange{IntType}
+    bit_basis([IntType], state::AbstractArray) -> UnitRange{IntType}
 
 Returns the UnitRange for basis in Hilbert Space of nbits qubits.
 If an array is supplied, it will return a basis having the same size
 with the first diemension of array.
 """
-basis(arg::Union{Int,AbstractArray}) = basis(Int, arg)
-basis(::Type{T}, nbits::Int) where {T<:Integer} = UnitRange{T}(0, 1 << nbits - 1)
-basis(::Type{T}, state::AbstractArray) where {T<:Integer} = UnitRange{T}(0, size(state, 1) - 1)
+bit_basis(arg::T) where T<:Integer = bit_basis(T, arg)
+bit_basis(arg::AbstractArray) = bit_basis(Int, arg)
```

Comment:
can we double-check all `Int` should be `Int64` explicitly? otherwise the behaviour will be very different on different system, unless you can throw error on general integer overflow correctly

```suggestion
bit_basis(arg::AbstractArray) = bit_basis(Int64, arg)
```